tests/test_metrics.py

```diff
@@ -179,6 +179,24 @@ def test_largebuffer(self):
 
         assert m.report()['key'] == 32768 + 1
 
+    def test_recent(self):
+        m = Metrics()
+        m2 = Metrics(shared=m.share())
+        m.add('test', SumMetric(1))
+        assert m.report() == {'test': 1}
+        assert m.report_recent() == {'test': 1}
+        m.clear_recent()
+        m.add('test', SumMetric(2))
+        assert m.report() == {'test': 3}
+        assert m.report_recent() == {'test': 2}
+        assert m2.report() == {'test': 3}
+        assert m2.report_recent() == {}
+        m2.add('test', SumMetric(3))
+        assert m2.report() == {'test': 6}
+        assert m.report() == {'test': 6}
+        assert m2.report_recent() == {'test': 3}
+        assert m.report_recent() == {'test': 2}
+
```

Comment:
I would add a `clear` and `clear_recent` for both `m` and `m2` at the end of this, 99.9% not needed, but doesn't heart.

Reply:
You mean with corresponding asserts to check state?

Reply:
Yes.